devito/passes/clusters/blocking.py

```diff
@@ -157,3 +160,85 @@ def decompose(ispace, d, block_dims):
     directions.update({bd: ispace.directions[d] for bd in block_dims})
 
     return IterationSpace(intervals, sub_iterators, directions)
+
+
+class Skewing(Queue):
+
+    """
+    Rebuild clusters with skewed expressions and skewed IterationSpace.
+
+    Notes
+    -----
+    This transformation is applying Loop skewing to derive the
+    wavefront method of execution of nested loops.  Loop skewing is
+    a simple transformation of loop bounds and is combined with loop
+    interchanging to generate the wavefront [1]_.
```

Comment:
so, "interchange" is part of skewing?

Reply:
No. `interchange` is part of wavefront method.
Wavefront needs skewing+interchange.

Wavefront with skewing+BLOCKING+interchange = even better wavefront.